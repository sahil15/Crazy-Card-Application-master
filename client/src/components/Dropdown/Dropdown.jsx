import React from "react";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import NativeSelect from "@material-ui/core/NativeSelect";

function Dropdown(props) {
  const handleChange = (e) => props.handleChange(e);
  const { value, title, menus, classDiv, required } = props;
  let optionItems = menus.map((menu) => <option key={menu}>{menu}</option>);

  return (
    <>
      <FormControl required={required} className={classDiv}>
        <InputLabel>{title}</InputLabel>
        <NativeSelect
          value={value}
          onChange={(e) => {
            handleChange(e, value);
          }}
        >
          {optionItems}
        </NativeSelect>
      </FormControl>
    </>
  );
}

export default Dropdown;
