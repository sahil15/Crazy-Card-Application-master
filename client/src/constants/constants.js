export const DETAILS = {
  textFields: [
    {
      name: "dob",
      label: "Birthday",
      type: "date",
      defaultValue: "2017-05-24",
    },
    {
      name: "address",
      label: "Address",
      minLength: 5,
      type: "text",
      helperText: "Address should be atleast 5 characters long",
    },
    {
      name: "city",
      label: "City",
      type: "text",
      minLength: 3,
      helperText: "City should be atleast 5 characters long",
    },
    {
      name: "postal",
      label: "Postal",
      type: "text",
      helperText: "Postal code should be atleast 5 characters long",
    },
    {
      name: "income",
      label: "Annual Income",
      type: "number",
      required: true,
      helperText: "Must be a number",
    },
  ],
};
export const CARD_DETAILS = [
  {
    name: "apr",
    text: "Representative % APR (variable)",
  },
  {
    text: "Balance Transfer Offer Duration",
    name: "balanceOfferDuration",
  },
  {
    text: "Purchase Offer Duration",
    name: "purchaseOfferDuration",
  },
  {
    text: "Credit Available",
    name: "credit",
  },
];

export const EMP_STATUS_LABEL = "Employment Status";
export const TITLE_LABEL = "Title";
export const TITLE_DROPDOWN = ["Mr", "Mrs", "Ms"];
export const EMP_STATUS_DROPDOWN = ["Full Time", "Part Time", "Student"];
export const HOME_BTN_TEXT = "Navigate to Home";
export const NO_CARD_TEXT = "No Card Selected";
export const INCOME_ERROR_MSG = "Must be a number";
export const INCOME_REQUIRED_MSG = "Required";
export const MIN_LENGTH_MSG = "Length must be atleast 3 characters long! ";

export const FETCH_CARDS_FAILED_MSG = "Fetch Cards Action Failed";
export const FETCH_AVAILABLE_CARDS_FAILED_MSG = "Fetch Available Card Failed";

export const baseURL = "http://localhost:3000";
