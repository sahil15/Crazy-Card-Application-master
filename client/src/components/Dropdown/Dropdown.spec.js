import React from "react";
import { shallow } from "enzyme";
import Dropdown from "./Dropdown";
import {TITLE_LABEL,TITLE_DROPDOWN } from "../../constants/constants";

describe("Tests for Dropdown", () => {

  it("Renders correctly", () => {
    const wrapper = shallow(
      <Dropdown classDiv="title"
      title={TITLE_LABEL}
      menus={TITLE_DROPDOWN}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
