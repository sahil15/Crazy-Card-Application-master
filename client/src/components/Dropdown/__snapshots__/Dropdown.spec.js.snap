// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Dropdown Renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="title"
  >
    <WithStyles(ForwardRef(InputLabel))>
      Title
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(NativeSelect))
      onChange={[Function]}
    >
      <option
        key="Mr"
      >
        Mr
      </option>
      <option
        key="Mrs"
      >
        Mrs
      </option>
      <option
        key="Ms"
      >
        Ms
      </option>
    </WithStyles(ForwardRef(NativeSelect))>
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;
